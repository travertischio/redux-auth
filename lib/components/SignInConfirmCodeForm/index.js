'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactIntl = require('react-intl');

var _immutable = require('redux-form/immutable');

var _reduxForm = require('react-form-fields/lib/TextField/reduxForm');

var _reduxForm2 = _interopRequireDefault(_reduxForm);

var _required = require('validators/lib/required');

var _messages = require('./messages');

var _messages2 = _interopRequireDefault(_messages);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SignInConfirmCodeForm = function SignInConfirmCodeForm(props) {
  var pristine = props.pristine,
      submitting = props.submitting;
  var formatMessage = props.intl.formatMessage;

  var codeLabel = formatMessage(_messages2.default.code);

  var handleSubmit = function handleSubmit(event) {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    props.handleSubmit();
  };

  return _react2.default.createElement(
    'form',
    { onSubmit: handleSubmit, noValidate: true },
    _react2.default.createElement(_immutable.Field, {
      name: 'code',
      id: 'code',
      label: codeLabel,
      placeholder: codeLabel,
      validate: [_required.required],
      component: _reduxForm2.default
    }),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(
        'button',
        {
          type: 'submit',
          disabled: pristine || submitting
        },
        'Sign In'
      )
    )
  );
}; /**
   *
   * SignInConfirmCodeForm
   *
   */

SignInConfirmCodeForm.propTypes = {
  handleSubmit: _propTypes2.default.func.isRequired,
  pristine: _propTypes2.default.bool.isRequired,
  submitting: _propTypes2.default.bool.isRequired,
  intl: _propTypes2.default.object.isRequired
};

exports.default = (0, _immutable.reduxForm)({
  form: 'signInConfirmCodeForm'
})((0, _reactIntl.injectIntl)(SignInConfirmCodeForm));