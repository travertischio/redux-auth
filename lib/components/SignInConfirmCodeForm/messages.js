'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactIntl = require('react-intl');

exports.default = (0, _reactIntl.defineMessages)({
  code: {
    id: 'app.components.SignInConfirmCodeForm.code',
    defaultMessage: 'Enter code'
  }
}); /*
     * SignInForm Messages
     *
     * This contains all the text for the SignInForm component.
     */