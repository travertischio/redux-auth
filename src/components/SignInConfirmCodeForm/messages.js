/*
 * SignInForm Messages
 *
 * This contains all the text for the SignInForm component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  code: {
    id: 'app.components.SignInConfirmCodeForm.code',
    defaultMessage: 'Enter code',
  },
});
