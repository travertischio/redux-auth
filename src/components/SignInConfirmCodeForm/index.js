/**
*
* SignInConfirmCodeForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import {
  Field,
  reduxForm,
} from 'redux-form/immutable';
import TextField from 'react-form-fields/lib/TextField/reduxForm';
import { required as requiredValidator } from 'validators/lib/required';
import messages from './messages';

const SignInConfirmCodeForm = (props) => {
  const {
    pristine,
    submitting,
  } = props;
  const { formatMessage } = props.intl;
  const codeLabel = formatMessage(messages.code);

  const handleSubmit = (event) => {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    props.handleSubmit();
  };

  return (
    <form onSubmit={handleSubmit} noValidate>
      <Field
        name="code"
        id="code"
        label={codeLabel}
        placeholder={codeLabel}
        validate={[requiredValidator]}
        component={TextField}
      />
      <div>
        <button
          type="submit"
          disabled={pristine || submitting}
        >
          Sign In
        </button>
      </div>
    </form>
  );
};

SignInConfirmCodeForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export default reduxForm({
  form: 'signInConfirmCodeForm',
})(injectIntl(SignInConfirmCodeForm));
