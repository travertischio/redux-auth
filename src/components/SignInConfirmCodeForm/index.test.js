import React from 'react';
import { createComponentWithIntl } from 'react-unit-testing-utils';

import SignInConfirmCodeForm from './';

describe('<SignInConfirmCodeForm />', () => {
  it('should render SignInForm', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<SignInConfirmCodeForm />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
