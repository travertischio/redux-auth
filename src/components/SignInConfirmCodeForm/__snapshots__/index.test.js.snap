// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInConfirmCodeForm /> should render SignInForm 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className=""
  >
    <div
      className="form-item"
    >
      <label
        htmlFor="code"
      >
        Enter code
      </label>
      <input
        className="textfield"
        id="code"
        name="code"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Enter code"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      disabled={true}
      type="submit"
    >
      Sign In
    </button>
  </div>
</form>
`;
